### PR TITLE
Add deferred weight reader pointer when initializing the graph

### PR DESCRIPTION
**Pull request by yinghai**
To add offline tensor support.

### CODE REVIEW

**Comment by yinghai** [comment]
@gcatron

**Review by gcatron — APPROVED** [review]

**Review by zrphercule — APPROVED** [review]
LGTM, please make sure where it is used being maintained.

File: foxi/onnxifi.h [hunk 1]
```diff
@@ -1607,6 +1613,9 @@ ONNXIFI_PUBLIC onnxStatus ONNXIFI_ABI
  * @param[out] graph - pointer to the opaque handle for the created ONNXIFI
  *                     graph. If the function fails, and this pointer is
  *                     non-NULL, the handle is initialized to NULL.
+ * @param[inout] deferredWeightReader - pointer to an opaque handle to populate
```
Comment (jackm321) [inline foxi/onnxifi.h:1616]: why is this inout?

Comment (yinghai) [reply foxi/onnxifi.h:1616]: Some of the method here is not constant. It will modified the underlying "workspace".

**Comment by gnaggnoyil** [comment]
I'm sorry, but wouldn't this commit(as well as 8015abb7) cause conflict between header declaration and source file definition conflict for `onnxInitGraph`? FYI both difinitions of onnxifi_dummy.c#L99 and onnxifi_wrapper.c#L757 are conflict with this modified function signature declaration

**Pull request merged**